lib/pages/clouds-page.tsx

```diff
@@ -0,0 +1,556 @@
+        type="range"
+        id={props.id}
+        min={props.min}
+        max={props.max}
+        value={props.value}
+        step={props.step}
+        onChange={(e) => props.onChange(parseFloat(e.target.value))}
+      />
+      <span>
+        {" "}
+        {props.value}
+        {props.valueSuffix}
+      </span>
+    </p>
+  );
+};
+
+/* SETTERS AND VARIABLES */
+const Clouds: React.FC<{}> = () => {
+  //const [stroke, setStroke] = useState(CLOUD_STROKE);
+  let [strokewidth, setStrokewidth] = useState(CLOUD_STROKEWIDTH);
+  //const [fill, setFill] = useState(CLOUD_FILL);
+  const [numElements, setnumElements] = useState(NUM_ELEMENTS);
+  //let [numClouds, setnumClouds] = useState(NUM_CLOUDS);
+	let numClouds = NUM_CLOUDS;
+  const [cloudDistance, setcloudDistance] = useState(CLOUD_DISTANCE);
+  let [numRows, setnumRows] = useState(NUM_ROWS);
+  const [rowDistance, setrowDistance] = useState(ROW_DISTANCE);
+  const [cloudSpeed, setcloudSpeed] = useState(CLOUD_SPEED);
+  const [parallaxSpeed, setparallaxSpeed] = useState(PARALLAX_SPEED);
+  const [parallaxSize, setparallaxSize] = useState(PARALLAX_SIZE);
+  const [rotationSpeed, setRotationSpeed] = useState(ROTATION_SPEED);
+  const [pulseDuration, setPulseDuration] = useState(PULSE_DURATION);
+  let [spacing, setSpacing] = useState(CLOUD_SPACING);
+  let [scaleValue, setScaleValue] = useState(CLOUD_SCALE); 
+  const [pulseValue, setPulseValue] = useState(CLOUD_PULSE); 
+  const [pulseminValue, setPulseminValue] = useState(CLOUD_PULSEMIN); 
+  const [compCtx, setCompCtx] = useState(createSvgCompositionContext());
+  const [useMask, setUseMask] = useState(false);
+	let [maskRadius, setMaskRadius] = useState(MASK_RADIUS); 
+	let [maskX, setMaskX] = useState(MASK_X); 
+	let [maskY, setMaskY] = useState(MASK_Y); 
+
+	
+	// simulate a Bezier pulse effect
+	let pulse2Value = pulseminValue;
+	let pulse3Value = pulseminValue;
+	let pulse4Value = pulseValue;
+	let pulseDiff = pulseValue - pulseminValue; // get the pulse difference
+	// if there is a difference then calculate the simulated Bezier effect
+	if (Math.abs(pulseDiff)>0) { 
+			pulse2Value = (pulseDiff / 7) + pulseminValue; /* pulse up to 1/7 difference  */
+			pulse3Value = (pulseDiff / 2) + pulseminValue; /* pulse up to 1/2 difference */
+			pulse4Value = pulseValue - (pulseDiff / 7); /* pulse up to 6/7 difference */
+	}
+
+
+ 
+	let rotationDuration = 0; //default to 0
+	if (rotationSpeed>0) {
+		rotationDuration = 8.5 - rotationSpeed; 
+	} else if (rotationSpeed<0) {
+		rotationDuration = 8.5 - (rotationSpeed * -1); // if rotationSpeed is negative then reverse direction
+	} 
+	let rotationAngle = -360;
+	if (rotationSpeed<0) {
+   rotationAngle = 360;
+	}
+
+	let screenWidth = SCREEN_WIDTH;
+
+	/* set default colors upon init */
+	if ((compCtx.background=="#858585") && (compCtx.stroke=="#000000") && (compCtx.fill=="#ffffff")) {
+	compCtx.background = BG_COLOR;
+	compCtx.stroke = CLOUD_STROKE;
+	compCtx.fill = CLOUD_FILL;
+	}
```

Comment:
This conditional depends on the default values of the composition context (which we may someday change) and also means that the user can never manually set the colors to black & white on gray...  instead, you may want to pass these values into `createSvgCompositionContext()` [above](https://github.com/mysticsymbolic/mysticsymbolic.github.io/pull/145/files#diff-f920a5ca8ec87cac7f94bf90ec15c9aa85b4cd54421636b21df8fd081668d4f5R101), so that the default colors are set to these values only upon initialization.

Reply:
@toolness - again - good catch! The reason I did this is that I wanted to change the default colors of the clouds - instead of black and white - to the light blue - which were the default colors of the waves - which I like.  Because the black and white were built into the defaults of \lib\colors.ts  as used in \lib\svg-symbol.tsx - I used this method to override the colors. It would be nice if we could pass default colors when calling the Color Randomizer widget.  I've been trying not to change many files outside of the Clouds Page.

Reply:
Right right--the thing is, the color widget doesn't actually manage the data for the actual colors--that's passed-in to it by your code, starting with the line that calls `createSvgCompositionContext()`.  That function takes an optional object that is the defaults--I can submit a PR to your PR that shows you what I mean!